velox/exec/benchmarks/Crc32Benchmark.cpp

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <folly/Benchmark.h>
+#include <folly/Random.h>
+#include <folly/hash/detail/ChecksumDetail.h>
+#include <array>
+#include <iostream>
+#include <random>
+#include <vector>
+
+#if defined(_MSC_VER) // Microsoft Visual Studio
+#include <intrin.h>
+#elif defined(__GNUC__) || defined(__clang__) // GCC or Clang
+#include <cpuid.h>
+#endif
+
+// Add the following definitions to allow Clion runs
+DEFINE_bool(gtest_color, false, "");
+DEFINE_string(gtest_filter, "*", "");
+
+namespace {
+
+bool isSSE42Supported() {
```

Comment:
There's `folly::CpuId` utility, better to use that: https://github.com/facebook/folly/blob/main/folly/CpuId.h#L180

Reply:
@yingsu00 Can we use either folly or xsimd for this?